app/models/concerns/attachmentable.rb

```diff
@@ -74,7 +74,7 @@ def obfuscate_file_name
     self.class.attachment_definitions.each_key do |attachment_name|
       attachment = send(attachment_name)
 
-      next if attachment.blank? || attachment.queued_for_write[:original].blank?
+      next if attachment.blank? || attachment.queued_for_write[:original].blank? || attachment.options[:preserve_files]
```

Comment:
How does the `preserve_files` thing work?

Reply:
`preserve_files` is set to `true` when `reprocess!` is called. It is not set in any other case. So it is a good signal for when we don't want to rename files. But it's actually a little problematic because I think it could leave the previous file around if the format (and extension) changes, and I plan on solving that before the PR is merged.

Reply:
Here is a comment from the `reprocess!` method:

> NOTE: Calling reprocess WILL NOT delete existing files. This is due to
> inconsistencies in timing of S3 commands. It's possible that calling
> #reprocess! will lose data if the files are not kept.

If the output filename is the same (such as when the file extension does not change), the file will be overwritten, and that's fine. I just need to add a special case when the `:original` file name changes and cause it to be deleted afterwards somehow.

Reply:
In the commit https://github.com/thoughtbot/paperclip/commit/5a06d85097c325905cd16b13c04d1d3e9d7ec0d9, it describes the problem as:

> Apparently, Amazon can not guarantee that PUT's called after DELETE's
is actually executed in that order.

In the case where we want to delete a file with a different extension, I think that doesn't matter, so should be fine to make it do that.